xrpl/utils/tx_parser/utils.py

```diff
@@ -0,0 +1,847 @@
+    """Normalize nodes.
+
+    Args:
+        metadata (Dict[str, Any]): Transctions metadata
+
+    Returns:
+        Iterable[NormalizedNode]: The normalized nodes
+    """
+    affected_nodes = metadata["meta"]["AffectedNodes"]
+    if not affected_nodes:
+        return []
+
+    return map(normalize_node, affected_nodes)
+
+
+"""
+#####################
+Utils balance_changes
+#####################
+"""
+
+
+class Balance:
+    """A accounts balance."""
+
+    def __init__(self: Any, counterparty: str, currency: str, value: str) -> None:
+        """
+        Args:
+            counterparty (str): Counterparty
+            currency (str): Currency
+            value (str): Value
+        """
+        self.counterparty = counterparty
+        self.currency = currency
+        self.value = value
+
+
+class TrustLineQuantity(object):
+    """Trust line quantity."""
+
+    def __init__(self: object, address: str, balance: Dict[str, str]) -> None:
+        """
+        Args:
+            address (str): Accounts address.
+            balance (Dict[str, str]): Accounts balance.
+        """
+        self.address = address
+        self.balance = Balance(
+            counterparty=balance["counterparty"],
+            currency=balance["currency"],
+            value=balance["value"],
+        )
+
+
+def _parse_value(value: Any) -> Decimal:
+    """Formats a balances value into a Decimal.
+
+    Args:
+        value (Any):
+            Balance.
+
+    Returns:
+        Decimal: Balance as Decimal.
+    """
+    if hasattr(value, "value"):  # issued currency amount
+        return Decimal(value.value)
+    else:  # XRP amount
+        return Decimal(value)
+
+
+def parse_final_balance(node: NormalizedNode) -> Union[Decimal, None]:
+    """Parse final balance.
+
+    Args:
+        node (NormalizedNode):
+            Normalized node.
+
+    Returns:
+        Union[Decimal, None]:
+            The balance value as Decimal.
+    """
+    if hasattr(node.new_fields, "Balance"):
+        return _parse_value(node.new_fields.Balance)
+    if hasattr(node.final_fields, "Balance"):
+        return _parse_value(node.final_fields.Balance)
+
+    return None
+
+
+def compute_balance_changes(node: NormalizedNode) -> Union[None, int, Decimal]:
+    """Compute balance changes.
+
+    Args:
+        node (NormalizedNode):
+            Normalized node.
+
+    Returns:
+        Union[None, int, Decimal]:
+            The parsed value.
+    """
+    value = None
+    if hasattr(node.new_fields, "Balance"):
+        value = _parse_value(node.new_fields.Balance)
+    if hasattr(node.previous_fields, "Balance") and hasattr(
+        node.final_fields, "Balance"
+    ):
+        value = _parse_value(node.final_fields.Balance) - _parse_value(
+            node.previous_fields.Balance
+        )
+
+    # if value is not 'None' or '0' return 'value' else 'None'.
+    return None if value is None else None if value == 0 else value
```

Comment:
```suggestion
     if( value and value != 0):
           return value
     else:
          return None
```
There's a lot of None's in this line and it's a bit confusing to read.